packages/ion-did-provider/src/ion-did-provider.ts

```diff
@@ -0,0 +1,97 @@
+import { IIdentifier, IKey, IService, IAgentContext, IKeyManager } from '@veramo/core'
+import { AbstractIdentifierProvider } from '@veramo/did-manager'
+import { keyUtils as secp256k1_keyUtils } from '@transmute/did-key-secp256k1'
+import * as ION from '@sphereon/ion-tools'
+import Debug from 'debug'
+
+const debug = Debug('sphereon:did-provider-ion')
+
+type IContext = IAgentContext<IKeyManager>
+
+/**
+ * {@link @veramo/did-manager#DIDManager} identifier provider for `did:ion` identifiers
+ * @public
+ */
+export class IonDIDProvider extends AbstractIdentifierProvider {
+  private defaultKms: string
+
+  constructor(options: { defaultKms: string }) {
+    super()
+    this.defaultKms = options.defaultKms
+  }
+
+  async createIdentifier(
+    { kms, options }: { kms?: string; options?: any },
+    context: IContext,
+  ): Promise<Omit<IIdentifier, 'provider'>> {
+    const key = await context.agent.keyManagerCreate({ kms: kms || this.defaultKms, type: 'Secp256k1' })
+    // convert to JWK (note, we should have a discussion about key dependencies here)
+    const publicJwk = secp256k1_keyUtils.publicKeyJwkFromPublicKeyHex(key.publicKeyHex)
+
+    const did = new ION.DID({
+      content: {
+        publicKeys: [
```

Comment:
Questions: Doesn;t it also need 2 other keys like for instance a recoveryKey?

Reply:
As discussed, created the following two tickets. Hopefully I wrote it down correctly

https://sphereon.atlassian.net/browse/WAL-204
https://sphereon.atlassian.net/browse/WAL-206